src/resilience/exec_space/openMP/OpenMPResSubscriber.hpp

```diff
@@ -127,39 +171,221 @@ struct CombineDuplicates: public CombineDuplicatesBase
+	return;
+      }
+      //No match found, all three executions return different number
+      success() = false;
+    }
+    // DMR has not failed over, 2 copies instantiated, 1 initialized
+    else{
+      //DMR combiner begin with no failover
+      if (check_equality.compare(copy[0](std::forward<Args>(its)...), original(std::forward<Args>(its)...))){
+	return;
+      }
+      success() = false;
+    }
+#else
+    //Main combiner begin
     for (int j = 0; j < 2; j ++) {
-      if (check_equality.compare(copy[j](i), original(i))) {
+      if (check_equality.compare(copy[j](std::forward<Args>(its)...), original(std::forward<Args>(its)...))) {
         return;
       }
     }
-    if (check_equality.compare(copy[0](i), copy[1](i))) {
-      original(i) = copy[0](i);  // just need 2 that are the same
+    if (check_equality.compare(copy[0](std::forward<Args>(its)...), copy[1](std::forward<Args>(its)...))) {
+      original(std::forward<Args>(its)...) = static_cast<typename View::value_type>(copy[0](std::forward<Args>(its)...));  // just need 2 that are the same
       return;
     }
     //No match found, all three executions return different number
-    success(0) = false;
+    success() = false;
+#endif
   }
 
-};
+  void Inject1D(){
+    //Sequential, on original.size
+    if ((original.size() != 1) && (ErrorInject::global_next_inject > original.size()))
+    {
+      ErrorInject::global_next_inject = ErrorInject::global_next_inject - original.size();
+    }
+
+    size_t next_inject = ErrorInject::global_next_inject;
+#if 0
+    double temp = global_error_settings->error_rate;
+    std::cout << "Error::error_rate is " << temp << std::endl;
+
+    std::cout << "ErrorInject::error_counter is " << ErrorInject::error_counter << std::endl;
+    std::cout << "next_inject is " << next_inject << std::endl;
+#endif
+    for (int j=0; j<=2; j++){
+      while (next_inject < original.size())
+      {
+        if (j==0){//Inject in the original if j is 0
+          original(next_inject) = static_cast<typename View::value_type>(2 * original(next_inject) + 2 * ErrorInject::random_gen());//generate using ()
+          ErrorInject::error_counter++;
+        }
+        else{//Else inject in one of the other two copies, copy[0] or copy[1]
+          copy[j-1](next_inject) = static_cast<typename View::value_type>(2 * copy[j-1](next_inject) + 2 * ErrorInject::random_gen());
+          ErrorInject::error_counter++;
+        }
+        next_inject = global_error_settings->geometric(ErrorInject::random_gen)+next_inject+1;
+
+      }
+      if(original.size() != 1){
+      next_inject = (next_inject) - (original.size());
+      }
+    }
+  }
+
+  void Inject2D(){
+    //This error injection works with 2D views, subtracts total extent from global_next_inject
+    size_t total_extent = original.extent(0) * original.extent(1);
```

Comment:
Let's merge `Inject2D()` and `Inject1D()` into one function as this will break with views of rank > 2 (also need to make the name consistent with the style which is snake case for function names).